### PR TITLE
Recursively handle subworkflow in dynamic

### PR DESCRIPTION
**Pull request by honnix**
# TL;DR
Recursively handle subworkflow in dynamic

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
Today subworkflow in dynamic task does not really work because we are not including all referenced task templates and propeller fails the compilation. This PR fixes that by recursively collecting all referenced task templates to include them in the dynamic job proto.

## Tracking Issue
_NA_

## Follow-up issue
_NA_

### CODE REVIEW

File: jflyte-utils/src/main/java/org/flyte/jflyte/utils/ProjectClosure.java [hunk 1]
```diff
@@ -343,8 +343,16 @@ && checkCycles(subWorkflowId, allWorkflows, beingVisited, visited))) {
...
+      List<Node> nodes, Map<WorkflowIdentifier, WorkflowTemplate> allWorkflows) {
+    return collectSubWorkflows(nodes, allWorkflows, Function.identity());
+  }
+
   public static Map<WorkflowIdentifier, WorkflowTemplate> collectSubWorkflows(
-      List<Node> rewrittenNodes, Map<WorkflowIdentifier, WorkflowTemplate> allWorkflows) {
+      List<Node> nodes,
+      Map<WorkflowIdentifier, WorkflowTemplate> allWorkflows,
+      Function<List<Node>, List<Node>> nodesRewriter) {
+    List<Node> rewrittenNodes = nodesRewriter.apply(nodes);
```
Comment (honnix) [inline jflyte-utils/src/main/java/org/flyte/jflyte/utils/ProjectClosure.java:355]: This is a bug fix because recursively collecting subworkflows doesn't work under dynamic task execution context.
